sensor/kubernetes/listener/resources/endpoints.go

```diff
@@ -240,12 +243,47 @@ func (m *endpointManagerImpl) OnDeploymentCreateOrUpdateByID(id string) {
 }
 
 func (m *endpointManagerImpl) onDeploymentCreateOrUpdate(deployment *deploymentWrap) {
+	data := m.endpointDataForDeployment(deployment)
 	updates := map[string]*clusterentities.EntityData{
-		deployment.GetId(): m.endpointDataForDeployment(deployment),
+		deployment.GetId(): data,
+	}
+	if deployment.GetName() == "sensor" {
+		m.updateHeritageData(data, true)
 	}
 	m.entityStore.Apply(updates, false, "OnDeploymentCreateOrUpdateByID")
 }
 
+func (m *endpointManagerImpl) updateHeritageData(data *clusterentities.EntityData, force bool) {
+	hm := m.entityStore.GetHeritageData()
+	if hm == nil {
+		return
+	}
+	if hm.HasCurrentSensorData() && !force {
+		return
+	}
+	var sensorContainerID, sensorPodIP string
+	sensorContainerIDs, sensorPodIPs := data.GetDetails()
+	if len(sensorContainerIDs) > 0 {
+		// Sort (if needed) as GetDetails is not guaranteed to return sorted data.
+		if !slices.IsSorted(sensorContainerIDs) {
+			slices.Sort(sensorContainerIDs)
+		}
+		sensorContainerID = sensorContainerIDs[0]
```

Comment:
Where do several container IDs come from, and why pick a random one?

Reply:
There is only one container ID. I am using the code you see here, because the data is stored in a map and I don't know how to query for "give me random key from a map of lenght==1". My solution to that problem is to convert to slice and get the 0-th element.
Sorting is an overkill here, but I wanted to be on the safe side in case something would change in the future.

Reply:
```go
  next, stop := iter.Pull(map.Keys())
  randomKey, ok := next()
  stop()
```

Reply:
```go
var randomKey string
for randomKey = range myMap {
    break
}
```

Reply:
Sorry, but I am still unsure what is the suggested change.
Do you want me to change the

```go
sensorContainerID = sensorContainerIDs[0]
```
into
```go
for sensorContainerID = range sensorContainerIDs {
	break
}
```
?

How is this clearer?

Reply:
I suggest you to not sort, and pick either of the options.

Reply:
I am sorry, but I am really hesitant to do this as I do not understand what you ask me to do.

1. How shall I not sort in:
```go
func (ed *EntityData) GetDetails() (containerIDs []string, podIPs []net.IPAddress) {
	containerIDs = slices.Sorted(maps.Keys(ed.containerIDs))
```
2. Why I cannot do `sensorContainerIDs[0]` in
```go
func (m *endpointManagerImpl) updateHeritageData(data *clusterentities.EntityData) error {
```

Maybe you can write an example explaining what you mean?

Reply:
I found this piece of code confusing:

```go
if !slices.IsSorted(sensorContainerIDs) {
	slices.Sort(sensorContainerIDs)
}
sensorContainerID = sensorContainerIDs[0]
```

Confusing, because the goal is to get the first container ID from a list of 1 (or 0?) element. But instead the code does:
1. `containerIDs = slices.Sorted(maps.Keys(ed.containerIDs))`
2. `if len(sensorContainerIDs) > 0`
3. `slices.IsSorted(sensorContainerIDs)`
4. `slices.Sort(sensorContainerIDs)`
5. `sensorContainerID = sensorContainerIDs[0]`

I thought a simpler solution would achieve the same goal with less effort. `GetDetails` could return the only container ID, or an empty string, without any sorting with just
```go
var containerID string
for containerID = range ed.containerIDs {
	break
}
```

Reply:
Basically, I don't understand the need for sorting of container IDs.

Reply:
> We are already searching for the container named sensor

I see only condition for deployment name and namespace.

> Because the data struct holding the container IDs may theoretically contain more IDs, I am sorting and getting the first one.

My point is that there is no sense in sorting, as the container IDs have no natural order. I'd suggest picking the ID of the container with name 'sensor'.